darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,374 @@
+"""
+Module for generating the Darshan job summary.
+"""
+import os
+import sys
+import glob
+import argparse
+import tempfile
+
+import numpy as np
+from fpdf import FPDF
+
+import darshan
+from darshan.experimental.plots.plot_dxt_heatmap import plot_heatmap
+from darshan.experimental.plots.plot_opcounts import plot_opcounts
+
+
+def truncate_label(label, label_width, max_width):
+    """
+    Truncates a label based on the maximum width of the label.
+
+    Parameters
+    ----------
+    label : str
+        The label to be altered.
+    label_width : float
+        The width of the label in points units.
+    max_width : float
+        The maximum width of the label in points units.
+
+    Returns
+    -------
+    truncated_label : str
+        The altered label with length less than the input maximum label width.
+    """
+    # calculate the number of characters per point
+    chars_per_pt = len(label) / label_width
+    # calculate the number of characters we can keep in the string based on
+    # the characters/point estimated above
+    # NOTE: this is an approximate value and likely changes with the string
+    # since helvetica is not a monospaced font
+    n_available_chars = chars_per_pt * max_width
+    # the first index is just the first half of the string
+    first_idx = int(n_available_chars / 2)
+    # the second index is the last half minus 3, adjusted for the 3 dots
+    second_idx = first_idx - 3
+    truncated_label = label[:first_idx] + "..." + label[-second_idx:]
+    return truncated_label
+
+
+class PDF(FPDF):
```

Comment:
Inheritance has been falling out of favor compared to composition for objected-oriented designs. Is `fpdf2` still recommending inheritance?

https://en.wikipedia.org/wiki/Composition_over_inheritance
https://en.wikipedia.org/wiki/Liskov_substitution_principle

Reply:
Yeah, in their [tutorial page](https://pyfpdf.github.io/fpdf2/Tutorial.html) they have an example that uses inheritance.

Reply:
Hmm, that's unfortunate, we may still be able to use composition if we want though.